pkg/types/validation/installconfig.go

```diff
@@ -770,9 +770,10 @@ func validateProxy(p *types.Proxy, c *types.InstallConfig, fldPath *field.Path)
 			v = strings.TrimSpace(v)
 			errDomain := validate.NoProxyDomainName(v)
 			_, _, errCIDR := net.ParseCIDR(v)
-			if errDomain != nil && errCIDR != nil {
+			ip := net.ParseIP(v)
+			if errDomain != nil && errCIDR != nil && ip == nil{
```

Comment:
shouldn't this be `or` comparison as the error message states?

Reply:
No, we enter here (and fail validation) if all 3 conditions report an error, i.e. the value being tested isn't any of the 3 supported types.